river/ensemble/boosting.py

```diff
@@ -119,3 +124,189 @@ def predict_proba_one(self, x):
+
+    Examples
+    --------
+
+    >>> from river.ensemble.boosting import BOLEBaseModel
+    >>> from river import datasets
+    >>> from river import ensemble
+    >>> from river import evaluate
+    >>> from river import drift
+    >>> from river import metrics
+    >>> from river import tree
+
+    >>> dataset = datasets.Elec2()
+
+
+    >>> model = ensemble.BOLEClassifier(
+    ...     model= BOLEBaseModel(model=tree.HoeffdingTreeClassifier(), drift_detection_method= drift.DDM()),
+    ...     n_models=10,
+    ...     seed=42
+    ... )
+
+    >>> metric = metrics.Accuracy()
+
+    >>> evaluate.progressive_val_score(dataset, model, metric)
+    Accuracy: 92.02%
+
+
+
+    References
+    ----------
+    [^1]: [Oza, N.C., 2005, October. Online bagging and boosting. In 2005 IEEE international conference on systems, man and cybernetics (Vol. 3, pp. 2340-2345). Ieee.](https://ti.arc.nasa.gov/m/profile/oza/files/ozru01a.pdf)
+
+    [^2]: R. S. M. d. Barros, S. Garrido T. de Carvalho Santos and P. M. Gonçalves Júnior, "A Boosting-like Online Learning Ensemble," 2016 International Joint Conference on Neural Networks (IJCNN), 2016, pp. 1871-1878, doi: 10.1109/IJCNN.2016.7727427.
+    """
+
+    def __init__(self, model: base.Classifier, n_models=10, seed: int = None, error_bound=0.5):
+        super().__init__(model=model, n_models=n_models, seed=seed)
+        self.error_bound = error_bound
+        self.orderPosition = np.arange(n_models)
+        self.instances_seen = 0
+
+    def learn_one(self, x, y):
+        self.instances_seen += 1
+
+        acc = np.zeros(self.n_models)
+        for i in range(self.n_models):
+            acc[i] = (
+                self.correct_weight[self.orderPosition[i]]
+                + self.wrong_weight[self.orderPosition[i]]
+            )
+            if acc[i] != 0:
+                acc[i] = self.correct_weight[self.orderPosition[i]] / acc[i]
+
+        # sort models in descending order by correct_weight / correct_weight + error_weight (best case insertion sort)
+        for i in range(1, self.n_models):
+            key_position = self.orderPosition[i]
+            key_acc = acc[i]
+            j = i - 1
+            while (j >= 0) and (acc[j] < key_acc):
+                self.orderPosition[j + 1] = self.orderPosition[j]
+                acc[j + 1] = acc[j]
+                j -= 1
+            self.orderPosition[j + 1] = key_position
+            acc[j + 1] = key_acc
+
+        correct = False
+        maxAcc = 0
+        minAcc = self.n_models - 1
+        lambda_poisson = 1
+
+        models = list(enumerate(self))
+        for i in range(len(models)):
+            if correct:
+                pos = self.orderPosition[maxAcc]
+                maxAcc += 1
+            else:
+                pos = self.orderPosition[minAcc]
+                minAcc -= 1
+
+            for _ in range(poisson(lambda_poisson, self._rng)):
+                models[pos][1].learn_one(x, y)
```

Comment:
You can do `self[pos][1].learn_one`, which avoids `models = list(enumerate(self))`

Reply:
the object is not subscriptable. Solved in last numpy removes and self.models[ind] commit. River has a self.models that returns the models.